### PR TITLE
NAT-PMP/UPNP 

### PR DESCRIPTION
**Pull request by brostosjoined**
This plugins autoport udp port 43210 to make the game accessible from the internet if upnp is available.

### CODE REVIEW

File: plugins/utilities/natpmp_upnp.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    # Try to add UDP port using NAT-PMP
+    import socket
+    import natpmp
+    from natpmp import NATPMPUnsupportedError, NATPMPNetworkError
+
+    try:
+        natpmp.map_port(
+            natpmp.NATPMP_PROTOCOL_UDP,
+            43210,
+            43210,
```
Comment (rikkolovescats) [inline plugins/utilities/natpmp_upnp.py:156]: do you think we should use `bascenev1.get_game_port()`? since game binds on a diff port if 43210 is taken.

**Comment by brostosjoined** [comment]
Let me do some research on nat-pmp

**Comment by brostosjoined** [comment]
Read some few RFC documents i thinks now everything is fine

**Review by rikkolovescats — APPROVED** [review]
thanks! :100:

**Comment by Loup-Garou911XD** [comment]
Cool stuff

**Pull request merged**